src/Native/System.Security.Cryptography.Native/openssl.c

```diff
@@ -848,3 +848,69 @@ int BioSeek(
 
     return BIO_seek(bio, ofs);
 }
+
+/*
+Function:
+NewX509Stack
+
+Used by System.Security.Cryptography.X509Certificates when needing to pass a collection
+of X509* to OpenSSL.
+
+Return values:
+A STACK_OF(X509*) with no comparator.
+*/
+STACK_OF(X509)*
+NewX509Stack()
+{
+    return sk_X509_new_null();
+}
+
+/*
+Function:
+PushX509StackField
+
+Used by System.Security.Cryptography.X509Certificates when needing to pass a collection
+of X509* to OpenSSL.
+
+Return values:
+1 on success
+0 on a NULL stack, or an error within sk_X509_push
+*/
+int
+PushX509StackField(
+    STACK_OF(X509)* stack,
+    X509* x509)
+{
+    if (!stack)
+    {
+        return 0;
+    }
+
+    return sk_X509_push(stack, x509);
+}
+
+/*
+Function:
+UpRefEvpPkey
+
+Used by System.Security.Cryptography.X509Certificates' OpenSslX509CertificateReader when
+duplicating a private key context as part of duplicating the Pal object
+
+Return values:
+1 on success
+0 if pkey is NULL, or if an invalid number of references remain, indicating that the key
+was already freed, or pkey is not pointing to a valid EVP_PKEY.
+*/
+int
+UpRefEvpPkey(
+    EVP_PKEY* pkey)
+{
+    if (!pkey)
+    {
+        return 0;
+    }
+
+    int i = CRYPTO_add(&pkey->references, 1, CRYPTO_LOCK_EVP_PKEY);
+
+    return (i > 1);
```

Comment:
Assert in the calling managed code that true was returned, or some sort of native assert?  (If "some sort of native assert" is the right answer, could you point it out to me? I don't know of any)